website/blog/2024-02-08-kubernetes-frontends/index.mdx

```diff
@@ -0,0 +1,195 @@
+identify which cluster it is placed in. Subsequently, it creates the REST config object, with which the clientset can be obtained.
+
+Similarly, when running outside the cluster, one needs to create the config object, but this config is read from the local kube config:
+
+```Go
+import (
+	"context"
+	"flag"
+	"path/filepath"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+func main() {
+	// get the passed (or default) kube config file path
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// do something with the clientset, e.g. getting all pods in the cluster:
+	// pods, err := clientset.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+}
+```
+
+Again, the Kubernetes Go client has us covered with a simple function to parse a kubeconfig file to get a config, which can then be used to create a clientset.
+
+When trying to run these simple examples, you will also come across one difference between these two approaches:
+Running the local tool is easier because you don't need to build an image, push it to a registry, and then pull it into the cluster.
+
+**Which one to choose?**
+
+Say you were to build your own Kubernetes UI in a similar fashion. When it comes to the decision of where the webserver of your tool should run, there are several things to consider:
+
+- **Distribution**: Running your tool inside the cluster means you have to build and distribute a docker image. On the contrary, you will have to distribute a native binary if you want users to install it on their machines. For both cases, there are lots of tools and resources online.
+- **Availability**: When your cluster is down for whatever reason, users might not be able to reach the tool hosted inside the cluster. This leads us to the next point:
+- **Onboarding User Experience**: This might be an edge case, but a locally hosted web tool is available before all of its components are installed in the cluster. This means you could implement some sort of UI onboarding flow for new users, making the tool easier to install and set up.
+- **Compatibility**: Multiple users of the same cluster could have different versions of your (locally hosted) tool installed. This could not happen if there is just one webserver running inside the cluster.
```

Comment:
```suggestion
- **Compatibility**: Multiple users of the same cluster could have different versions of your (locally hosted) tool installed. This can not happen if there is just one webserver running inside the cluster.
```